src/tests/test_utils/common_test_utils/include/common_test_utils/node_builders/constant.hpp

```diff
@@ -0,0 +1,79 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#pragma once
+
+#include "common_test_utils/ov_tensor_utils.hpp"
+#include "openvino/core/node.hpp"
+#include "openvino/op/constant.hpp"
+#include "ov_models/utils/data_utils.hpp"
+
+namespace ov {
+namespace test {
+namespace utils {
+
+namespace deprecated {
+// Legacy implementation
+// Remove after transition to new one
+template <typename T>
+std::shared_ptr<ov::Node> make_constant(const ov::element::Type& type,
+                                        const std::vector<size_t>& shape,
+                                        const std::vector<T>& data,
+                                        bool random = false,
+                                        T up_to = 10,
+                                        T start_from = 1,
+                                        const int seed = 1) {
+#define makeNode(TYPE)                                                                                              \
+    case TYPE:                                                                                                      \
+        if (random) {                                                                                               \
+            return std::make_shared<ov::op::v0::Constant>(                                                          \
+                type,                                                                                               \
+                shape,                                                                                              \
+                NGraphFunctions::Utils::generateVector<TYPE>(ov::shape_size(shape),                                 \
+                                                             ov::element_type_traits<TYPE>::value_type(up_to),      \
+                                                             ov::element_type_traits<TYPE>::value_type(start_from), \
+                                                             seed));                                                \
+        } else {                                                                                                    \
+            if (std::is_same<T, fundamental_type_for<TYPE>>::value) {                                               \
+                return std::make_shared<ov::op::v0::Constant>(type, shape, data);                                   \
+            } else {                                                                                                \
```

Comment:
As I can see, in most cases use of `make_constant` doesn't require random data. So, this `if/else` construction may be replaced with inplace `ov::op::v0::Constant::create` instead of calling `make_constant`

Let's try to use production code and decrease usage of helper

Reply:
Unfortunately, we cannot do it automatically. It requires some manual efforts.
So I suggest to merge it as is to speed up api 2.0 migration and change it later.